maestro-android/src/main/java/dev/mobile/maestro/receivers/LocaleSettingReceiver.kt

```diff
@@ -0,0 +1,113 @@
+package dev.mobile.maestro.receivers
+
+import android.app.Activity
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import dev.mobile.maestro.handlers.LocaleSettingHandler
+import org.apache.commons.lang3.LocaleUtils
+import java.util.*
+
+class LocaleSettingReceiver : BroadcastReceiver(), HasAction {
+    override fun onReceive(context: Context, intent: Intent) {
+        var language = intent.getStringExtra(LANG)
+        var country = intent.getStringExtra(COUNTRY)
+
+        if (language == null || country == null) {
+            Log.w(TAG, "It is required to provide both language and country, for example: " +
+                    "am broadcast -a dev.mobile.maestro --es lang ja --es country JP")
+            Log.i(TAG, "Set en-US by default.")
+            language = "en"
+            country = "US"
+        }
+
+        var locale = Locale(language, country)
+
+        Log.i(TAG, "Obtained locale: $locale")
+
+        val script = intent.getStringExtra(SCRIPT)
+        if (script != null) {
+            Locale.Builder().setLocale(locale).setScript(script).build().also { locale = it }
+        }
+
+        if (!LocaleUtils.isAvailableLocale(locale)) {
+            val approximateMatchesLc = matchLocales(language, country)
+
+            if (approximateMatchesLc.isNotEmpty() && script.isNullOrBlank()) {
+                Log.i(
+                    TAG,
+                    "The locale $locale is not known. Selecting the closest known one ${approximateMatchesLc[0]} instead"
+                )
+                locale = approximateMatchesLc[0]
+            } else {
+                val approximateMatchesL = matchLocales(language)
+                if (approximateMatchesL.isEmpty()) {
+                    Log.e(
+                        TAG,
+                        "The locale $locale is not known. Only the following locales are available: ${LocaleUtils.availableLocaleList()}"
+                    )
+                } else {
+                    Log.e(
+                        TAG,
+                        "The locale $locale is not known. " +
+                                "The following locales are available for the $language language: $approximateMatchesL" +
+                                "The following locales are available altogether: ${LocaleUtils.availableLocaleList()}"
+                    )
+                }
+                resultCode = RESULT_LOCALE_NOT_VALID
+                resultData = "Failed to set locale $locale, the locale is not valid"
+                return
+            }
+        }
+
+        try {
+            LocaleSettingHandler(context).setLocale(locale)
+            Log.i(TAG, "Set locale: $locale")
+            resultCode = RESULT_SUCCESS
+            resultData = locale.toString()
+        } catch (e: Exception) {
+            Log.e(TAG, "Failed to set locale", e)
```

Comment:
Curious, Do we know of any known/unknown exceptions that can happen here?

Reply:
since we're using a fixed list of supported locales , the probability that it gets here is close to 0. but it is still worth to propagate this error, since it is hard to test that every language + region combination would work in all the API levels we support.

if exception is related to reflection APIs those might be thrown:

ClassNotFoundException,
NoSuchMethodException,
InvocationTargetException,
IllegalAccessException,
NoSuchFieldException